src/Calculator.ts

```diff
@@ -345,3 +339,196 @@ function addSDE(report: R4.IMeasureReport, measureURL: string, result: Execution
+  [PopulationType.IPP]: 'Initial Population',
+  [PopulationType.DENOM]: 'Denominator',
+  [PopulationType.DENEX]: 'Denominator Exclusion',
+  [PopulationType.DENEXCEP]: 'Denominator Exception',
+  [PopulationType.NUMER]: 'Numerator',
+  [PopulationType.NUMEX]: 'Numerator Exclusion',
+  [PopulationType.MSRPOPL]: 'Measure Population',
+  [PopulationType.MSRPOPLEX]: 'Measure Population Exclusion',
+  [PopulationType.OBSERV]: 'Measure Observation'
+};
+
+// http://build.fhir.org/ig/HL7/cqf-measures/ValueSet-aggregate-method.html
+function aggregate(aggregation: string, observations: number[]) {
+  // Note: add break for any non-returning case
+  switch (aggregation) {
+    case AggregationType.SUM:
+      // sum	Sum	The measure score is determined by adding together the observations derived from the measure population.
+      return observations.reduce((accumulator, currentValue) => accumulator + currentValue);
+    case AggregationType.AVERAGE:
+      // average	Average	The measure score is determined by taking the average of the observations derived from the measure population.
+      return observations.reduce((accumulator, currentValue) => accumulator + currentValue) / observations.length;
+    case AggregationType.MEDIAN:
+      // median	Median	The measure score is determined by taking the median of the observations derived from the measure population.
+      return median(observations);
+    case AggregationType.MIN:
+      // minimum	Minimum	The measure score is determined by taking the minimum of the observations derived from the measure population.
+      return Math.min(...observations);
+    case AggregationType.MAX:
+      // maximum	Maximum	The measure score is determined by taking the maximum of the observations derived from the measure population.
+      return Math.max(...observations);
+    case AggregationType.COUNT:
+      // count	Count	The measure score is determined as the number of observations derived from the measure population.
+      return observations.length;
+    default:
+      // should not default, TODO: add error?
+      return observations.length;
+  }
+}
+
+function median(observations: number[]) {
+  const sorted = observations.sort(function (a, b) {
+    return a - b;
+  });
+  if (sorted.length % 2 === 0) {
+    return 0.5 * (sorted[sorted.length / 2 - 1] + sorted[sorted.length / 2]);
+  } else {
+    return sorted[sorted.length / 2];
+  }
+}
+
+function populationTotal(group: R4.IMeasureReport_Group, type: PopulationType) {
+  return (
+    group.population?.find(pop => {
+      return pop.code?.coding && pop.code.coding.length > 0 && pop.code?.coding[0].code === type;
+    })?.count || 0.0
+  );
+}
+
+function createMeasureScore(
+  measure: R4.IMeasure,
+  group: R4.IMeasureReport_Group,
+  detail: DetailedPopulationGroupResult,
+  report: R4.IMeasureReport
+) {
+  const scoringCode =
+    measure.scoring?.coding?.find(c => c.system === 'http://hl7.org/fhir/measure-scoring')?.code || '';
```

Comment:
Maybe we should throw an error here if no measure scoring is defined?

Reply:
I added the error under the default for the switch below (which is what will happen with an empty string), so this will catch both unfound and unexpected scoring codes